File: flutter/macos/Classes/MedeaJasonPlugin.m
#import "MedeaJasonPlugin.h"

@implementation MedeaJasonPlugin
+ (void)registerWithRegistrar:(nonnull id<FlutterPluginRegistrar>)registrar {
    FlutterMethodChannel* channel = [FlutterMethodChannel
        methodChannelWithName:@"medea_jason"
              binaryMessenger:[registrar messenger]];
    MedeaJasonPlugin* instance = [MedeaJasonPlugin alloc];
    MedeaJasonPlugin* finalInstance =
        [instance initWithChannel:channel:[registrar messenger]];
    [registrar addMethodCallDelegate:finalInstance channel:channel];
}

// Handles provided FlutterMethodCall.
- (void)handleMethodCall:(nonnull FlutterMethodCall*)methodCall
                  result:(nonnull FlutterResult)result {
    result(FlutterMethodNotImplemented);
}

// Initializes this FlutterWebRTCPlugin with a FlutterMethodChannel.
- (instancetype)initWithChannel:(FlutterMethodChannel*)
                        channel:(NSObject<FlutterBinaryMessenger>*)messenger {
    self = [super init];
    return self;
}
@end

File: flutter/macos/Classes/MedeaJasonPlugin.swift
import Cocoa
import FlutterMacOS

public class MedeaJasonPlugin: NSObject, FlutterPlugin {
  public static func register(with registrar: FlutterPluginRegistrar) {
    let channel = FlutterMethodChannel(name: "medea_jason", binaryMessenger: registrar.messenger)
    let instance = MedeaJasonPlugin()
    registrar.addMethodCallDelegate(instance, channel: channel)
  }

  public func handle(_ call: FlutterMethodCall, result: @escaping FlutterResult) {
    switch call.method {
    case "getPlatformVersion":
      result("macOS " + ProcessInfo.processInfo.operatingSystemVersionString)
    default:
      result(FlutterMethodNotImplemented)
    }
  }
}
